src/hpc/threadapi.c

```diff
@@ -374,13 +374,9 @@ static int GetThreadID(const char * funcname, Obj thread)
     else if (TNUM_OBJ(thread) == T_THREAD) {
         return ThreadID(thread);
     }
-
-
-#define AS_STRING(s) #s
-    const char *msg = "must be thread object or an integer between 0 and "
-                AS_STRING(MAX_THREADS-1);
-#undef AS_STRING
-    RequireArgumentEx(funcname, thread, NICE_ARGNAME(thread), msg);
+    RequireArgumentEx(funcname, thread, NICE_ARGNAME(thread),
+                      "must be a thread object or an integer between 0 and "
+                      "MAX_THREADS - 1");
```

Comment:
Is MAX_THREADS a gap level variable people can query? If not, might be nice to put something here people can check.

Reply:
It is currently only defined in the kernel, but could be exposed easily if needed:

```
Obj FuncMAX_THREADS()
{
    return INTOBJ_INT(MAX_THREADS);
}
```

Reply:
I added `ExportAsConstantGVar(MAX_THREADS);` now, as `MAX_THREADS` is a compile time constant. That said, I am not sure putting it into the global namespace is a good idea? One alternative would be to put into `GAPInfo.KernelInfo`.

BTW, note before my change, the error message also said `MAX_THREADS`, i.e., it never printed a number (it was meant to do so, but the code was buggy and untested. Oh, and of course it was written by me *sigh*).